src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowStateAction.java

```diff
@@ -0,0 +1,109 @@
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.rest;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.flowframework.common.FlowFrameworkFeatureEnabledSetting;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.transport.GetWorkflowStateAction;
+import org.opensearch.flowframework.transport.GetWorkflowStateRequest;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_ID;
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_URI;
+import static org.opensearch.flowframework.common.FlowFrameworkSettings.FLOW_FRAMEWORK_ENABLED;
+
+/**
+ * Rest Action to facilitate requests to get a workflow status
+ */
+public class RestGetWorkflowStateAction extends BaseRestHandler {
+
+    private static final String GET_WORKFLOW_STATE_ACTION = "get_workflow_state";
+    private static final Logger logger = LogManager.getLogger(RestGetWorkflowStateAction.class);
+    private FlowFrameworkFeatureEnabledSetting flowFrameworkFeatureEnabledSetting;
+
+    /**
+     * Instantiates a new RestGetWorkflowStateAction
+     * @param flowFrameworkFeatureEnabledSetting Whether this API is enabled
+     */
+    public RestGetWorkflowStateAction(FlowFrameworkFeatureEnabledSetting flowFrameworkFeatureEnabledSetting) {
+        this.flowFrameworkFeatureEnabledSetting = flowFrameworkFeatureEnabledSetting;
+    }
+
+    @Override
+    public String getName() {
+        return GET_WORKFLOW_STATE_ACTION;
+    }
+
+    @Override
+    protected BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+
+        try {
+            if (!flowFrameworkFeatureEnabledSetting.isFlowFrameworkEnabled()) {
+                throw new FlowFrameworkException(
+                    "This API is disabled. To enable it, update the setting [" + FLOW_FRAMEWORK_ENABLED.getKey() + "] to true.",
+                    RestStatus.FORBIDDEN
+                );
+            }
+
+            // Validate content
+            if (request.hasContent()) {
+                throw new FlowFrameworkException("No request body present", RestStatus.BAD_REQUEST);
```

Comment:
I assume same issue as @dbwiddis mentioned above https://github.com/opensearch-project/flow-framework/pull/273/files#r1423261483.

Can be fixed here or separately when fixing the issue.

Reply:
@dbwiddis we can address this at a later time when actioning that issue